File: 229-majority-element-ii/majority-element-ii.java
import java.util.*;

class Solution {
    public List<Integer> majorityElement(int[] nums) {
        Map<Integer, Integer> hashmap = new HashMap<>();
        List<Integer> res = new ArrayList<>();

        for (int n : nums) {
            hashmap.put(n, hashmap.getOrDefault(n, 0) + 1);
        }

        for (Map.Entry<Integer, Integer> entry : hashmap.entrySet()) {
            if (entry.getValue() > nums.length / 3) {
                res.add(entry.getKey());
            }
        }

        return res;
    }
}

File: 229-majority-element-ii/majority-element-ii.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        count = defaultdict(int)

        for n in nums:

            count[n] += 1

            if len(count) <= 2:
                continue

            new_count = defaultdict(int)

            for n , val in count.items():

                if val > 1:
                    new_count[n] = val - 1
            count = new_count
        res = []

        for num in count:
            if nums.count(num) > len(nums) //3:
                res.append(num)
        return res


        